### PR TITLE
One off hosted runners

### PR DESCRIPTION
**Pull request by crohr**
Note: I (@crohr) will perform the merge, since I need to make sure the corresponding GitHub App is in place.

* ~10min to complete the full test suite
* As many concurrent runs as needed: hosted runners are started on-demand, and automatically destroyed after ONE job run.
* Allows runs from pull requests, since runners are now recycled after every run.
* Don't start workflow if changes only target documentation folders.
* Add support for uploading screenshots for failures to S3
* Commit the runtime logs, so that parallel_test can balance the test groups (might require some update every few month)

Runners currently use m5zn instance types, which are the fastest cloud instances available (even faster than the OVH bare metal runner we were using previously). They use spot instances so the cost is ~0.3$/hour of test, i.e. 0.1$ per test suite (20min) assuming we stay under the 10min mark for both unit and feature tests.

Example test run: https://github.com/opf/openproject/runs/1910690212?check_suite_focus=true

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -233,7 +233,7 @@ group :test do
   gem 'json_spec', '~> 1.1.4'
   gem 'shoulda-matchers', '~> 4.5', require: nil
 
-  gem 'parallel_tests', '~> 3.1'
+  gem 'parallel_tests'
```
Comment (oliverguenther) [inline Gemfile]: Why did you remove the version pinning here? Usually we try to pin to a minor version to be able to safely upgrade gems

Comment (crohr) [reply Gemfile]: Fixed it :)

**Review by oliverguenther — APPROVED** [review]
👍 from my side
